### ISSUE TITLE
Control over generation length?

### ISSUE THREAD

**Issue by GrahamboJangles** [salesforce/ctrl#23]
Is there an option to control generation length, if not, are there plans on this?

**Comment by umhau** [comment]
[This](https://github.com/salesforce/ctrl/blob/7b77fdefeced5322dc83f62c746a019655457a3d/generation.py#L277) looks like the point in the text generation loop that you could insert a check to control the length of the string.

**Comment by keskarnitish** [comment]
Is there anything you wish to do that isn't captured by the `--generate_num` flag?

**Comment by umhau** [comment]
For myself, I was hoping the solution here would have a way to end the generation at the end of a sentence, rather than at a word count.  Any good way to control it that way?

**Comment by keskarnitish** [comment]
You could detect if the `idx2word` of the generated token has a period as a proxy for end-of-sentence. This will falsely trigger on tokens like `U.S.A` though.  A better heuristic, depending on the use case, could be if `\n` is being predicted. 

@bmccann what do you think?

**Comment by bmccann** [comment]
Newlines would be easy, but might not cut it, e.g. if the desired functionality is to auto-complete a sentence at a time.

**Comment by umhau** [comment]
I haven't checked what the behavior is in most of the control codes, but for unattended sentence generation, it seems like newlines could work.  Just test to make sure there's a sentence preceeding it, I guess.  It looks like we went through the same thought process re: period detections.  

I found a couple tools for boundary detection [here](https://stackoverflow.com/questions/56416641/separate-texts-into-sentences-nltk-vs-spacy). Looks like they might be compute-intensive as well, but maybe not too much.
